### PR TITLE
KPI, Status, ViewportManager, useTimeSeriesData and useViewport

### PR DESCRIPTION
**Pull request by diehbria**
### KPI and Status
further development on making these production ready, adding tests, cleaning the code and fixing minor UX issues. These components are not yet exported from the library.

### Viewport manager, useTimeSeriesData, useViewport
Added new features to make development of IoT App Kit components adhere to react best practices and simplify the overall DX. These features are not yet exported from the library.

### Documentation
- Removed the internal packages that should not be used by users of the library. The home readme.md page is designed for users of the library, rather than internal development.

- Added documentation for new features, but have not llinked them to the main README.md so they are only discoverable by searching the code repo. I have included in the title and at the start of the page that these features are in development.

- Removed the section from custom data sources about utilizing TimeSeriesDataModule. Long term, we should get arid of this and use websockets.

- Various improvements and fixes on documentation

## Summary
- Iterate on KPI and Status component
- Create `<ViewportManager />` component, to manage viewports and synchronize across groups
- create `useViewport` react-hook to consume viewports provided by the `ViewportManager`
- Iterate on `useTimeSeriesData` hook to handle viewports, styles
- Add in missing tests (still more to come)
- Documentation on viewport manager
- Documentation on useViewport
- Documentation on useTimeSeriesData
- Storybook updates for KPI, Status, and ViewportManager
- refactor some of the components into a `component` folder to separate them from other various util and common scripts

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Comment by diehbria** [comment]
need to revise to remove the documentation changes until the work is exposed from the package

**Review by TheEvilDev — COMMENTED** [review]
Some comments haven't gotten through the whole review yet

File: docs/useTimeSeriesData.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+## Basic example usage
+
+The simplest use of the `useTimeSeriesData` is to provide a query for time series data, which will return you the time series data streams
+for the queried data, as well as the currently active viewport in the viewport group.
+Learn more about viewport groups at the [Viewport Manager documentation](https://github.com/awslabs/iot-app-kit/tree/main/docs/ViewportManager.md).
+
+```
+    import { useTimeSeriesData } from '@iot-app-kit/react-components';
+    const { dataStreams, viewport, setViewport } = useTimeSeriesData({ query });
```
[the remark this answers is not in the document]
Comment (diehbria) [reply docs/useTimeSeriesData.md]: good point, i have updated it and removed it from the `useTimeSeriesData` hook

File: docs/ViewportManager.md [hunk 2]
```diff
@@ -1,11 +1,156 @@
...
+### useViewport API Summary
+
+The `useViewport` takes no inputs, and returns an object containing the following fields:
+
+### viewport
+
+The current viewport for the viewport group subscribed to. Will be undefined if the hook is used outside of the context of a `<ViewportManager />`.
+Will also be undefined if the associated viewport group has no associated viewport.
+
+Type: Object or undefined
```
[the remark this answers is not in the document]
Comment (diehbria) [reply docs/ViewportManager.md]: Have to think more about this one, might change things around in a future revision before I expose these functionalities as public. For the very simple use cases i want to make the barrier to get started as simple as possible and am hesitant to require anything more than just a single stand alone visualizaiton

File: docs/ViewportManager.md [hunk 3]
```diff
@@ -1,11 +1,156 @@
 # Viewport manager
-
-The viewport manager is an exposed utility of `@iot-app-kit/core` which allows IoT App Kit components to synchronize their viewport within their
-specified viewport group.
+The `<ViewportManager />` is a component of IoT App Kit which enables the management of viewports between all of your IoT App Kit components, enabling use cases such as synchronizing viewports across components in one or more groups, and providing the ability
```
[the remark this answers is not in the document]
Comment (diehbria) [reply docs/ViewportManager.md]: Yes i agree, I want to tackle this at some point this year and have a fuller 'iot app kit' doc site with live examples

File: docs/useTimeSeriesData.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+// jsx
+<ViewportManager initialViewport={{ duration: '5m' }}>
+  <CustomVisualization 
+    query={
+      query.timeSeriesData({ 
+        assets: [{ 
+          assetId: 'windmill', 
+          properties: [{ propertyId: 'rpm' }]
+        }]
+      })
```
[the remark this answers is not in the document]
Comment (diehbria) [reply docs/useTimeSeriesData.md:48]: Going to keep it consistent with the rest of the documentation for the time being but I do want to take on a bigger task later to revamp the documentation with storybook (or something similar) with code that executes

File: docs/useTimeSeriesData.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    import { useTimeSeriesData } from '@iot-app-kit/react-components';
+    const { dataStreams, viewport, setViewport } = useTimeSeriesData({ query });
+```
+
+A common use case of this hook is to construct an IoT App Kit visualizaiton, which may look simillar to the below example:
+```
+//jsx
+const CustomVisualization = () => {
+    const { dataStreams, viewport, setViewport } = useTimeSeriesData({ query });
+    return <VisualizationComponent dataStreams={dataStreams} viewport={viewport} setViewport={setViewport} />
```
[the remark this answers is not in the document]
Comment (diehbria) [reply docs/useTimeSeriesData.md]: refactored to keep `useViewport` and `useTimeSeriesData` having 'single responsibility' like you suggest

**Review by jmbuss — APPROVED** [review]

**Pull request merged**